lib/framework/model.dart

```diff
@@ -13,37 +14,60 @@ class IconModel {
 }
 
 class BackgroundImage {
-  BackgroundImage(this._source, {BoxFit? fit, Alignment? alignment})
-      : _fit = fit ?? BoxFit.cover,
-        _alignment = alignment ?? Alignment.center;
+  BackgroundImage(
+    this._source, {
+    BoxFit? fit,
+    Alignment? alignment,
+    dynamic fallback,
+  })  : _fit = fit ?? BoxFit.cover,
+        _alignment = alignment ?? Alignment.center,
+        _fallback = fallback;
 
   final String _source;
   final BoxFit _fit;
   final Alignment _alignment;
+  final dynamic _fallback;
+
+  DecorationImage getImageAsDecorated(ScopeManager? scopeManager) {
+    final Widget? fallbackWidget = _fallback != null
+        ? scopeManager?.buildWidgetFromDefinition(_fallback)
+        : null;
 
-  DecorationImage get asDecorationImage {
     ImageProvider imageProvider;
     if (Utils.isUrl(_source)) {
       imageProvider = NetworkImage(_source);
     } else {
       imageProvider = AssetImage(Utils.getLocalAssetFullPath(_source));
     }
     return DecorationImage(
-        image: imageProvider, fit: _fit, alignment: _alignment);
+      image: imageProvider,
+      fit: _fit,
+      alignment: _alignment,
+      onError: (_, __) =>
```

Comment:
@vusters Here, the DecorationImage onError doesn't have a Widget as a return type. Need to find another solution for getting the image as a decorated one.